yolapy/resources/partner.py

```diff
@@ -0,0 +1,70 @@
+class PartnerResourceMixin(object):
+
+    """Methods for managing Partner resources."""
+
+    def create_partner(self, **attributes):
+        """Create a partner.
+
+        ```
+        partner = yola.create_partner(
+            id='COMPANY_ID',
+            name='Company Name',
+            parent_partner_id='PARENT_ID',
+            properties={'name': 'value'})
+        partner['name'] # => 'Company Name'
+        ```
+
+        See <https://wl.qa.yola.net/partners/> for available properties.
+        """
+        return self.post(self._partner_path(), data=attributes).json()
+
+    def get_partner(self, partner_id):
+        """Return details for a particular partner.
+
+        ```
+        partner = yola.get_partner('PARTNER_ID')
+        partner['name'] # => 'Company Name'
+        ```
+        """
+        return self.get(self._partner_path(partner_id)).json()
+
+    def delete_partner(self, partner_id):
+        """Delete a partner.
+
+        `yola.delete_partner('PARTNER_ID')`
+        """
+        self.delete(self._partner_path(partner_id))
+
+    def update_partner(self, partner_id, **attributes):
+        """Update a partner.
+
+        `yola.update_partner('PARTNER_ID', name='New name')`
+        """
+        return self.patch(
+            self._partner_path(partner_id), data=attributes).json()
+
+    def list_partners(self, **options):
+        """Return paginated list of partners.
+
+        `yola.list_partners()`
+
+        Example response:
+
+        ```
+        {
+            'count': 999,
+            'previous': None,
+            'next': 'https://wl.qa.yola.net/pr/partners/?page=2',
+            'results': [{...}, {...}, ...]
+        }
+        ```
+
+        You can pass `page_size` and `page` as keyword arguments:
```

Comment:
@thomasw: I think we're in agreement.

> That said, I would expect the one to one client method / API method mappings to land and be reviewed in isolation of higher level abstractions. - @thomasw 

That's what I meant by: "In our clients, the service classes are super thin wrappers over API calls, and that's all we wanted to include in this (for now)"

> The higher level abstractions should be added in subsequent pulls based on anticipated usage.

That's what I meant by: "Higher-level abstractions like pagination helpers, models, etc. would be up to the consumers of this (which may be us, but not yet, and not in this pull)."

I guess since I'll be using this for bottomline as my next priority issue the consumer will _definitely_ be us, so I probably should have just replied with: "I agree, we'll add that in a subsequent pull."